clients/drcachesim/tests/burst_traceopts.cpp

```diff
@@ -209,9 +209,53 @@ gather_trace(const std::string &tracer_ops, const std::string &out_subdir)
     return post_process(out_subdir);
 }
 
+static void
+reg_id_set_unit_tests()
+{
+    reg_id_set_t set;
+    assert(set.begin() == set.end());
+    // Test non-GPR.
+    auto insert_res = set.insert(IF_X86_ELSE(DR_REG_XMM0, DR_REG_Q0));
+    assert(insert_res.first == set.end());
+    assert(!insert_res.second);
+    assert(set.begin() == set.end());
+    // Test adding a GPR.
+    insert_res = set.insert(DR_REG_START_GPR + 1);
+    assert(insert_res.first != set.end());
+    assert(insert_res.second);
+    assert(set.begin() != set.end());
+    // Test find not-present.
+    auto find_res = set.find(DR_REG_START_GPR);
+    assert(find_res == set.end());
+    // Test find present.
+    find_res = set.find(DR_REG_START_GPR + 1);
+    assert(find_res != set.end());
+    assert(*find_res == DR_REG_START_GPR + 1);
+    // Test erase with two entries.
+    insert_res = set.insert(DR_REG_START_GPR + 4);
+    assert(insert_res.second);
+    auto iter = set.begin();
+    bool found_next = false;
+    while (iter != set.end()) {
+        if (*iter == DR_REG_START_GPR + 1)
+            iter = set.erase(iter);
+        else {
+            if (*iter == DR_REG_START_GPR + 4)
+                found_next = true;
+            ++iter;
+        }
+    }
+    assert(found_next);
+    assert(set.find(DR_REG_START_GPR + 1) == set.end());
+    assert(set.find(DR_REG_START_GPR + 4) != set.end());
+    assert(*set.find(DR_REG_START_GPR + 4) == DR_REG_START_GPR + 4);
+}
+
 int
 main(int argc, const char *argv[])
 {
+    reg_id_set_unit_tests();
```

Comment:
Will it be useful to move this to a separate test file? I'm not aware how other such custom classes are tested currently.

Reply:
core unit tests have a framework, but there aren't really any parallels to this outside the core.  This test covers trace opts, and this class is used for the opts, so it is a logical fit as part of the same set of tests.  With the upstream CI having limited resources, and in particular Appveyor being rather slow and each new separate executable launching slowly on Windows and affecting developer productivity, it's better for project usability to have tests combined into fewer executables (like googletest or other frameworks do, combining related tests into a single executable).  IMHO placing this here as part of the related tests is no loss in modularity and a win in the CI.

Reply:
Ack, thanks for explaining.

Reply:
Actually a separate file but linked into this test would work and may be nicer for separating out the class test -- I will try that.

Reply:
There is also clients/drcachesim/tests/drcachesim_unit_tests.cpp.  I moved this code into clients/drcachesim/tests/reg_id_set_unit_tests.cpp.  It is nicer that way.  +1.